### ISSUE TITLE
Aws::Delete needed for PooledThreadExecutor OverflowPolicy::REJECT_IMMEDIATELY?

### ISSUE THREAD

**Issue by albumcover** [aws/aws-sdk-cpp#891]
```

bool PooledThreadExecutor::SubmitToThread(std::function<void()>&& fn)
{
    //avoid the need to do copies inside the lock. Instead lets do a pointer push.
    std::function<void()>* fnCpy = Aws::New<std::function<void()>>(POOLED_CLASS_TAG, std::forward<std::function<void()>>(fn));

    {
        std::lock_guard<std::mutex> locker(m_queueLock);

        if (m_overflowPolicy == OverflowPolicy::REJECT_IMMEDIATELY && m_tasks.size() >= m_poolSize)
        {
            Aws::Delete(fnCpy); //FIXME: needed?
            return false;
        }

        m_tasks.push(fnCpy);
    }

    m_sync.Release();
  
    return true;
}
```

**Comment by marcomagdy** [comment]
Yes, that is a bug. Thanks for reporting it.

I always wondered if people are using this overflow policy. Can you share your use case?

**Comment by albumcover** [comment]
@marcomagdy we are using the default policy... just came across this line of code

**Comment by marcomagdy** [comment]
Fixed since v1.5.12
